nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/cf/JMSConnectionFactoryProvider.java

```diff
@@ -185,23 +185,31 @@ public void disable() {
      * service configuration. For example, 'channel' property will correspond to
      * 'setChannel(..) method and 'queueManager' property will correspond to
      * setQueueManager(..) method with a single argument.
-     * <p>
+     * <br>
      * There are also few adjustments to accommodate well known brokers. For
      * example ActiveMQ ConnectionFactory accepts address of the Message Broker
      * in a form of URL while IBMs in the form of host/port pair (more common).
      * So this method will use value retrieved from the 'BROKER_URI' static
      * property 'as is' if ConnectionFactory implementation is coming from
-     * ActiveMQ and for all others (for now) the 'BROKER_URI' value will be
+     * ActiveMQ or Tibco. For all others (for now) the 'BROKER_URI' value will be
      * split on ':' and the resulting pair will be used to execute
      * setHostName(..) and setPort(..) methods on the provided
-     * ConnectionFactory. This may need to be maintained and adjusted to
-     * accommodate other implementation of ConnectionFactory, but only for
-     * URL/Host/Port issue. All other properties are set as dynamic properties
-     * where user essentially provides both property name and value, The bean
-     * convention is also explained in user manual for this component with links
-     * pointing to documentation of various ConnectionFactories.
+     * ConnectionFactory. An exception to this if the ConnectionFactory
+     * implementation is coming from IBM MQ and multiple brokers are listed,
+     * in this case setConnectionNameList(..) method is executed.
+     * This may need to be maintained and adjusted to accommodate other
+     * implementation of ConnectionFactory, but only for URL/Host/Port issue.
+     * All other properties are set as dynamic properties where user essentially
+     * provides both property name and value, The bean convention is also
+     * explained in user manual for this component with links pointing to
+     * documentation of various ConnectionFactories.
```

Comment:
Thanks for elaborating the Javadocs. Very helpful to understand how the same property can be used differently based on the target MQ broker systems.

The sad thing is, these helpful information will not be seen by general NiFi users from NiFi UI usage.

I think this is a good timing to adding more on 'Additional Details' page for this component. As a user I'd like to see following information laid out nicely in the additional doc page:

- Target MQ System: i.e. ActiveMQ, IBM, Tibco ... etc
- MQ ConnectionFactory Implementation
- Broker URI: Especially for the IBM case where single queue manager instance and multiple ones.

We can link to external doc pages for each vender specific information from there.

How do you think?

Reply:
Your welcome, it's my pleasure . Indeed

Agreed, it should be available to the end-user. I will add this information and links to external docs

Reply:
I tried to add this information and elaborate on the Additional Details' page. What do you think?

Reply:
Thanks for adding the details docs, that is very helpful!